### PR TITLE
Open channel scenario added

### PR DESCRIPTION
**Pull request by CosminNechifor**
Almost all cases were implemented except for the open channel with 0 tokens.
Do not merge.
Fixes: #2559

### CODE REVIEW

File: raiden/tests/scenarios/open_channel_scenario.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      - parallel:
+          name: "Checking for different response codes"
+          tasks:
+            ## Try to open a channel without having enough tokens for the initial deposit
+            - open_channel: {from: 0, to: 2, total_deposit: 20, expected_http_status: 402}
+            ## Try to open a channel with himself
+            - open_channel: {from: 0, to: 0, total_deposit: 20, expected_http_status: 409}
+            ## Try to open a channel with the same address partner and check for expected response
+            - open_channel: {from: 0, to: 1, total_deposit: 10, expected_http_status: 409}
+            ## Try to open a channel with 0 tokens for the initial deposit
```
[the remark this answers is not in the document]
Comment (CosminNechifor) [reply raiden/tests/scenarios/open_channel_scenario.yaml]: Ok

File: raiden/tests/scenarios/open_channel_scenario.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    name: "Open channel scenarios"
+    tasks:
+      - serial:
+          name: "Stop node 4"
+          tasks:
+            - stop_node: 4
+      - parallel:
+          name: "Opening channels"
+          tasks:
+            ## Try to open 5 channels
```
[the remark this answers is not in the document]
Comment (CosminNechifor) [reply raiden/tests/scenarios/open_channel_scenario.yaml]: On: Keyboard shortcuts.

Off: got myself a new mechanical keyboard and can't stop typing.

File: raiden/tests/scenarios/open-channel-diferent-scenarios.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+settings:
```
Comment (palango) [inline raiden/tests/scenarios/open-channel-diferent-scenarios.yaml]: s/diferent/different in the file name

File: raiden/tests/scenarios/open-channel-diferent-scenarios.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            - stop_node: 3
+      - parallel:
+          name: "Testing different cases"
+          tasks:
+            # Try to open a channel with himself
+            - open_channel: {from: 0, to: 0, total_deposit: 10, expected_http_status: 409}
+            # Try to open a channel with the partner and check for expected response
+            - open_channel: {from: 0, to: 1, total_deposit: 10, expected_http_status: 409}
+            # Try to open a channel with 0 tokens for the initial deposit
+            # - open_channel: {from: 2, to: 3, total_deposit: 0, expected_http_status: 201}
```
Comment (palango) [inline raiden/tests/scenarios/open-channel-diferent-scenarios.yaml]: Why is this commented out?

File: raiden/tests/scenarios/open-channel-happy-scenario.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+settings:
```
Comment (palango) [inline raiden/tests/scenarios/open-channel-happy-scenario.yaml]: What's the goal of this scenario? Everything it tests is alreday tested in the other scenario. I would just remove it.

File: raiden/tests/scenarios/open-channel-diferent-scenarios.yaml (hunk 3 above)
Comment (CosminNechifor) [reply raiden/tests/scenarios/open-channel-diferent-scenarios.yaml]: Thanks, will rename it soon.

File: raiden/tests/scenarios/open-channel-diferent-scenarios.yaml (hunk 4 above)
Comment (CosminNechifor) [reply raiden/tests/scenarios/open-channel-diferent-scenarios.yaml]: It fails and I need to investigate this separately.

File: raiden/tests/scenarios/open-channel-happy-scenario.yaml (hunk 5 above)
Comment (CosminNechifor) [reply raiden/tests/scenarios/open-channel-happy-scenario.yaml]: Ok will remove it.

File: tools/scenario-player/scenario_player/tasks/channels.py [hunk 6]
```diff
@@ -19,7 +19,7 @@ def _request_params(self):
             ],
         )
         total_deposit = self._config.get('total_deposit')
-        if total_deposit:
+        if total_deposit or total_deposit == 0:
```
Comment (palango) [inline tools/scenario-player/scenario_player/tasks/channels.py]: IMO it would be cleaner to write `if total_deposit is not None`

Comment (CosminNechifor) [reply tools/scenario-player/scenario_player/tasks/channels.py]: Yea. Will do that after I test it out. I also thought that it makes no sens the or.

Comment (palango) [reply tools/scenario-player/scenario_player/tasks/channels.py]: The problem here is that `0` is falsy in python

**Comment by LefterisJP** [comment]
@palango @CosminNechifor what's holding this back?

**Comment by palango** [comment]
@LefterisJP COsmin had some problems running the scenarios due to spammed ropsten. This is the last thing required before merging.

**Pull request merged**